File: ohara-kafka/src/test/scala/com/island/ohara/kafka/connector/SimpleRowSourceConnector.scala
package com.island.ohara.kafka.connector

/**
  * Used for testing.
  */
class SimpleRowSourceConnector extends RowSourceConnector {
  private[this] var config: TaskConfig = _

  override def _start(config: TaskConfig): Unit = {
    this.config = config
  }

  override def _taskClass(): Class[_ <: RowSourceTask] = classOf[SimpleRowSourceTask]

  override def _taskConfigs(maxTasks: Int): Seq[TaskConfig] = Seq.fill(maxTasks)(config)

  override def _stop(): Unit = {}
}

File: ohara-kafka/src/test/java/com/island/ohara/kafka/connector/SimpleRowSourceConnector.java
package com.island.ohara.kafka.connector;

import java.util.Collections;
import java.util.List;

/** Used for testing. */
public class SimpleRowSourceConnector extends RowSourceConnector {

  private TaskConfig config = null;

  @Override
  protected Class<? extends RowSourceTask> _taskClass() {
    return SimpleRowSourceTask.class;
  }

  @Override
  protected List<TaskConfig> _taskConfigs(int maxTasks) {
    return Collections.nCopies(maxTasks, config);
  }

  @Override
  protected void _start(TaskConfig config) {
    this.config = config;
  }

  @Override
  protected void _stop() {}
}
